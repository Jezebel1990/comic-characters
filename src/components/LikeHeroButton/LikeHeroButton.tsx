// @ts-ignore
import Dislike from '../../assets/like.png'
// @ts-ignore
import Like from '../../assets/heartClick3x.png'
import React from 'react'
interface LikeHeroButtonProps {
  isLiked: boolean
  onClick: () => void
}

export function LikeHeroButton({ isLiked, onClick }: LikeHeroButtonProps) {
  const handleClick = () => {
    onClick()
  }

  return (
    <button
      onClick={handleClick}
      style={{ background: 'none', border: 'none', cursor: 'pointer' }}
    >
      <img
        src={isLiked ? Like : Dislike}
        alt={isLiked ? 'Liked' : 'Not Liked'}
        width={14}
        height={14}
      />
    </button>
  )
}
