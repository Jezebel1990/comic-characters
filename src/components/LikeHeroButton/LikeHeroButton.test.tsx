import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { LikeHeroButton } from './LikeHeroButton'

describe('<LikeHeroButton />', () => {
  it('should render the correct image based on isLiked prop', () => {
    // Renderiza o componente com isLiked = false
    const { container } = render(
      <LikeHeroButton isLiked={false} onClick={() => {}} />
    )

    // Verifica se a imagem "Not Liked" (dislike) é exibida
    expect(screen.getByAltText('Not Liked')).toBeInTheDocument()

    // Renderiza o componente com isLiked = true
    render(<LikeHeroButton isLiked={true} onClick={() => {}} />)

    // Verifica se a imagem "Liked" é exibida
    expect(screen.getByAltText('Liked')).toBeInTheDocument()

    // Gerar snapshot do botão
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should call onClick when button is clicked', () => {
    const handleClick = jest.fn()

    // Renderiza o componente com isLiked = false e mock de onClick
    render(<LikeHeroButton isLiked={false} onClick={handleClick} />)

    // Encontra o botão e dispara o evento de clique
    const button = screen.getByRole('button')
    fireEvent.click(button)

    // Verifica se a função onClick foi chamada
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('should toggle isLiked state when clicked', () => {
    let isLiked = false
    const handleClick = () => {
      isLiked = !isLiked
    }

    // Renderiza o componente com isLiked = false
    render(<LikeHeroButton isLiked={isLiked} onClick={handleClick} />)

    // Encontra o botão e dispara o evento de clique
    const button = screen.getByRole('button')
    fireEvent.click(button)

    // Verifica se o estado foi alterado
    expect(isLiked).toBe(true) // O estado deve mudar para true após o clique
  })
})
