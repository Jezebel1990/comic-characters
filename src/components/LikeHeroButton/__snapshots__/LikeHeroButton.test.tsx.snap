// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeHeroButton /> should render the correct image based on isLiked prop 1`] = `
<button
  style="background: none; cursor: pointer;"
>
  <img
    alt="Not Liked"
    height="14"
    src="test-file-stub"
    width="14"
  />
</button>
`;
